account_financial_report/models/account_move_line.py

```diff
@@ -1,11 +1,29 @@
 # Copyright 2019 ACSONE SA/NV (<http://acsone.eu>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-
-from odoo import api, models
+from odoo import api, fields, models
 
 
 class AccountMoveLine(models.Model):
     _inherit = "account.move.line"
 
+    analytic_account_ids = fields.Many2many(
+        "account.analytic.account", compute="_compute_analytic_account_ids", store=True
```

Comment:
You must avoid this new stored field that will be triggered on module installation. Imagine if you already have millions of journal items...

Why is this needed?

Reply:
@pedrobaeza 
The analytic_account_id field has been changed to analytic_distribution which is a JSON.
I've created this new field because, in order to filter by Analytical Account, I don't know any other way to do it with a JSON field in a domain. And that created field has to be stored to be able to use it in a domain.

This is used in [reports/general_ledger.py#L278](https://github.com/OCA/account-financial-reporting/blob/117d2e5976f414b5b41768e933f71c545c1d5a7f/account_financial_report/report/general_ledger.py#L278)

You know some way?

Thinking now, i think that we could use the analytic lines `analytic_line_ids` field to filter using the domain `('analytic_line_ids.account_id', 'in, 'xxx')`. But these lines are not created until the invoice is published, and I don't know if this is a problem for a report